src/transactions/TransactionFrame.cpp

```diff
@@ -225,9 +225,10 @@ TransactionFrame::getNumOperations() const
 Resource
 TransactionFrame::getResources(bool useByteLimitInClassic) const
 {
-    int64_t txSize = xdr::xdr_size(mEnvelope);
     if (isSoroban())
     {
+        auto txSize = static_cast<int64_t>(
```

Comment:
Why did this move inside the if? Seems cleaner to get the size just once like before.

Reply:
We don't use the size in the final `else` case (non-soroban, non-useByteLimitInClassic), so I moved the computation into the cases where we are using it. But I agree its cleaner to only have the computation in one place.

Reply:
looks like txSize is still calculated twice

Reply:
Reverted.